### PR TITLE
Picking correct depth/stencil binding point relevant to attachment aspect

### PR DESCRIPTION
**Pull request by ElArtista**

### CODE REVIEW

**Comment by ElArtista** [comment]
PR has dependency to #3531, will rebase it after it gets merged.

File: src/backend/gl/src/device.rs [hunk 1]
```diff
@@ -1114,12 +1114,22 @@ impl d::Device<B> for Device {
             }
 
             if let Some(depth_stencil) = subpass.depth_stencil {
+                let aspects = match &attachments[depth_stencil] {
```
Comment (kvark) [inline src/backend/gl/src/device.rs:1117]: A little pet peeve of mine (sorry!) - could we please not match on a reference?
There is no need to take advantage of the extra sugar here, we'd rather want explicit code, i.e. `match attachments[depth_stencil] {...}`.

File: src/backend/gl/src/device.rs [hunk 2]
```diff
@@ -1114,12 +1114,22 @@ impl d::Device<B> for Device {
...
+                    n::ImageView::Texture { sub, .. } => &sub.aspects,
+                    n::ImageView::Renderbuffer { aspects, .. } => aspects,
+                };
+
+                let attachment = [
```
Comment (kvark) [inline src/backend/gl/src/device.rs:1122]: let's not try to be too fancy here
```rust
if aspects == Aspects::DEPTH { glow::DEPTH_ATTACHMENT }
else if aspects == Aspects::STENCIL { glow::STENCIL_ATTACHMENT }
else { glow::DEPTH_STENCIL_ATTACHMENT }
```

File: src/backend/gl/src/native.rs [hunk 3]
```diff
@@ -253,7 +256,10 @@ impl SwapchainImage {
                 num_levels: 1,
                 num_layers: 1,
             },
-            view: ImageView::Renderbuffer(renderbuffer),
+            view: ImageView::Renderbuffer {
+                raw: renderbuffer,
+                aspects: format::Aspects::empty(),
```
Comment (kvark) [inline src/backend/gl/src/native.rs:261]: any reason to not use `::COLOR` here?

Comment (ElArtista) [reply src/backend/gl/src/native.rs:261]: I'm matching the aspect above in the format :)

File: src/backend/gl/src/device.rs (hunk 1 above)
Comment (ElArtista) [reply src/backend/gl/src/device.rs:1117]: Done.

File: src/backend/gl/src/device.rs (hunk 2 above)
Comment (ElArtista) [reply src/backend/gl/src/device.rs:1122]: Ok!

File: src/backend/gl/src/native.rs (hunk 3 above)
Comment (kvark) [reply src/backend/gl/src/native.rs:287]: oh my!

**Review by kvark — APPROVED** [review]
bors r+

**Pull request merged**